### PR TITLE
Inc 61/date picker test

### PR DESCRIPTION
**Pull request by Viarcinski**
Covered the "date picker" component with unit tests.

### CODE REVIEW

File: @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx [hunk 1]
```diff
@@ -0,0 +1,203 @@
+import { render, cleanup, screen, fireEvent } from '@testing-library/react'
+import { MainDatePicker } from '../DatePicker'
+describe('date picker renders correctly', () => {
+  it('date picker render', () => {
+    const { asFragment } = render(<MainDatePicker />)
+    expect(asFragment()).toMatchSnapshot()
```
Comment (TURP1) [inline @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx:6]: you should add all props variants of Datepicker to snapshot

File: @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx [hunk 2]
```diff
@@ -0,0 +1,203 @@
+import { render, cleanup, screen, fireEvent } from '@testing-library/react'
+import { MainDatePicker } from '../DatePicker'
+describe('date picker renders correctly', () => {
```
Comment (TURP1) [inline @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx:3]: afterEach(cleanup)

File: @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx
Comment (TURP1) [inline @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx:1]: you should definitely read how to name tests. Your descriptions are not very clear.
start of test description ` it('...')`
is called that for a reason

File: @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    render(<MainDatePicker value="01/01/1900" />)
+
+    fireEvent.click(screen.getByRole('button'))
+    expect(screen.getByTitle('Previous month')).toBeDisabled()
+  })
+
+  it('"Next month" button is disabled on last month of last year', async () => {
+    render(<MainDatePicker value="12/31/2099" />)
+
+    fireEvent.click(screen.getByRole('button'))
```
Comment (TURP1) [inline @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx:166]: screen.getByRole('button') 
you can use variables , try not to repeat yourself

File: @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    expect(screen.getByPlaceholderText('MM/DD/YYYY').getAttribute('value')).toBe('10/24/2000')
+  })
+
+  it('clicking on the "next month" button changes the month', async () => {
+    render(<MainDatePicker value="10/10/2000" />)
+
+    fireEvent.click(screen.getByRole('button'))
+    fireEvent.click(screen.getByTitle('Next month'))
+    fireEvent.click(screen.getAllByRole('gridcell', { name: '20' })[0])
+    expect(screen.getByPlaceholderText('MM/DD/YYYY').getAttribute('value')).toBe('11/20/2000')
```
Comment (TURP1) [inline @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx:126]:
```
await waitFor(() => {
    expect(screen.getByPlaceholderText('MM/DD/YYYY').getAttribute('value')).toBe('11/20/2000');
  });
```
look in all async tests this issue

File: @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+  })
+
+  it('clicking on the "previous month" button changes the month', async () => {
+    render(<MainDatePicker value="10/10/2000" />)
+
+    fireEvent.click(screen.getByRole('button'))
+    fireEvent.click(screen.getByTitle('Previous month'))
+    fireEvent.click(screen.getAllByRole('gridcell', { name: '20' })[0])
```
Comment (TURP1) [inline @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx:134]: need an empty line after all fireEvents / userEvents

File: @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+    render(<MainDatePicker value="12/31/2099" />)
+
+    const datePickerElement = screen.getByPlaceholderText('MM/DD/YYYY')
+
+    expect(datePickerElement).toHaveAttribute('value')
+    expect(datePickerElement.parentElement).not.toHaveClass('Mui-error')
+    expect(datePickerElement.getAttribute('value')).toBe('12/31/2099')
+  })
+
+  it('higher out of border value displays with error', () => {
```
Comment (TURP1) [inline @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx:71]: what about positive border values ?

**Review by TURP1 — APPROVED** [review]
need minor changes

**Review by TURP1 — CHANGES_REQUESTED** [review]
need minor changes

File: @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx
Comment (Viarcinski) [reply @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx:1]: done

File: @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx (hunk 1 above)
Comment (Viarcinski) [reply @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx:32]: done

File: @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx (hunk 4 above)
Comment (Viarcinski) [reply @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx:165]: done

File: @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx (hunk 5 above)
Comment (Viarcinski) [reply @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx:173]: done

File: @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx (hunk 3 above)
Comment (Viarcinski) [reply @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx]: done

**Review by TURP1 — APPROVED** [review]
GJ

File: @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx (hunk 1 above)
Comment (TURP1) [reply @ui/ui-kit/DatePicker/__test__/Datepicker.test.tsx:32]: look how to create only 1 testcase for all snapshots with cycle , check the Button.test

**Review by ilich01 — APPROVED** [review]

**Pull request merged**